projects/plugins/jetpack/extensions/shared/external-media/sources/google-photos/index.js

```diff
@@ -1,14 +1,41 @@
+import { useEffect, useState } from 'react';
+import MediaLoadingPlaceholder from '../../media-browser/placeholder';
 import { MediaSource } from '../../media-service/types';
 import withMedia from '../with-media';
 import GooglePhotosAuth from './google-photos-auth';
 import GooglePhotosMedia from './google-photos-media';
+import GooglePhotosPickerButton from './google-photos-picker-button';
 
 function GooglePhotos( props ) {
-	if ( ! props.isAuthenticated ) {
+	const { isAuthenticated, pickerSession, createPickerSession, getPickerStatus } = props;
+	const [ pickerFeatureEnabled, setPickerFeatureEnabled ] = useState( null );
+	const isPickerSessionAccurate = pickerSession !== null && ! ( 'code' in pickerSession );
+
+	useEffect( () => {
+		getPickerStatus().then( feature => {
+			feature && setPickerFeatureEnabled( feature.enabled );
+		} );
+	}, [ getPickerStatus ] );
+
+	useEffect( () => {
+		if ( pickerFeatureEnabled && isAuthenticated && ! isPickerSessionAccurate ) {
+			createPickerSession();
+		}
+	}, [ pickerFeatureEnabled, isPickerSessionAccurate, isAuthenticated, createPickerSession ] );
+
+	if ( pickerFeatureEnabled === null ) {
+		return <MediaLoadingPlaceholder />;
+	}
+
+	if ( ! isAuthenticated ) {
 		return <GooglePhotosAuth { ...props } />;
 	}
 
-	return <GooglePhotosMedia { ...props } />;
+	if ( pickerFeatureEnabled && ! pickerSession?.mediaItemsSet ) {
+		return <GooglePhotosPickerButton { ...props } />;
+	}
+
+	return <GooglePhotosMedia pickerFeatureEnabled { ...props } />;
```

Comment:
Does this means `pickerFeatureEnabled` is true all the time?

Reply:
Nice catch. I reproduced the issue on the regular flow. 👍

Reply:
Fixed. Thanks!!!